### ISSUE TITLE
Move transport-specific handlers to sub-packages in `jsonrpc`

### ISSUE THREAD

**Issue by joshklop** [NethermindEth/juno#985]
`jsonrpc` implements the (transport-agnostic) JSON-RPC v2 spec. Dependents should not need to rely on transport code in `net/http` and `nhooyr.io/websocket` without explicit requesting it.

**Comment by omerfirmak** [comment]
Is this still relevant?

**Comment by joshklop** [comment]
Yes. I was planning on doing this after #1389, but ended up focusing on other things in the meantime.

**Comment by joshklop** [comment]
Noticed while working on this: we are using the [same namespace, subsystem, and name](https://github.com/NethermindEth/juno/blob/27c1ac20c794d8c7c95c71daa0669415b628669f/node/metrics.go#L78-L91) for all RPC method counters, across [all servers](https://github.com/NethermindEth/juno/blob/27c1ac20c794d8c7c95c71daa0669415b628669f/node/http.go#L80). Do we need to aggregate them somehow?
